### DISCUSSION TITLE
How to (if possible) remove rounded borders in annotating highlight?

### DISCUSSION THREAD

**Discussion by dennisheyuxi** [sumatrapdfreader/sumatrapdf#4516] [General]
The curved border just looks off, especially when it comes to multi-line content. (Looks like a giant obese worm). Any way to do this in the advanced options? Thanks guys.

**Comment by GitHubRulesOK** [comment]
Not without customising the core code as it is a MuPDF "Feature" they introduced a while back.
Here it was in 1.18 (4 years ago in 2020) and still is in 1.25 unsure if there was a different method before that but it is what SumatraPDF inherits
![image](https://github.com/user-attachments/assets/b3d3cd14-41a4-4a0a-ad4e-cc347a004ede)
![image](https://github.com/user-attachments/assets/16d8daf4-ec3b-405a-89da-c1811628bc25)

In theory you could edit the additions which are several KB for one small box ! the offender is the sequence with line endings  `m c l c` needs change to `m l l l` by remove some of the circular bloat so
```
16 0 obj
<</Type/XObject/Subtype/Form/BBox[42.453508 751.255 186.8585 787.27297]/Matrix[1 0 0 1 0 0]/Resources<</ExtGState<</H<</Type/ExtGState/BM/Multiply>>>>>>/Length 159>>
stream
/H gs
1 1 0 rg
50 753.256 m
42.453508 760.8025 42.453508 777.72549 50 785.272 c
179.31201 785.272 l
186.8585 777.72549 186.8585 760.8025 179.31201 753.256 c
f

endstream
endobj
```
adapted to look like this (NOTE there are 2 boxes) so not just two changes needed there would be others to consider

![image](https://github.com/user-attachments/assets/9aa93829-2b13-4ef8-9eb0-790d1ea1dddb)

```16 0 obj
<</Type/XObject/Subtype/Form/BBox[42.453508 751.255 186.8585 787.27297]/Matrix[1 0 0 1 0 0]/Resources<</ExtGState<</H<</Type/ExtGState/BM/Multiply>>>>>>/Length 159>>
stream
/H gs
1 1 0 rg
50 753.256 m
50 785.272 l
179.31201 785.272 l
179.31201 753.256 l
f

endstream
endobj
```

**Reply by dennisheyuxi** [reply]
Hey mate, thanks so much for your reply. However, where do I place the code? Is it under `SumatraPDF-settings.txt` or elsewhere? Again, thanks, this could be a life saver.

**Reply by GitHubRulesOK** [reply]
There is no code that is me hand editing the adapted PDF in MS Notepad whilst watching in SumatraPDF (not an easy way to work)
This question has been asked before long ago in old forum but not raised as a problem (https://github.com/sumatrapdfreader/sumatrapdf/issues/1594) There have been 2 different shapes but both with rounded ends https://github.com/sumatrapdfreader/sumatrapdf/discussions/4386 

There is an open issue mainly since it affects areas and vertical text https://github.com/sumatrapdfreader/sumatrapdf/issues/2489 
Could perhaps be adjusted in code as per my culling the curves as shown above but that would be a lot of coding in the source files as provided by Artifex.

My edit of PDF can probably be done easily in any GUI Editor like Acrobat Pro or might even be possible in NotePad++ but have not tested how good that is for editing PDF. There are command line editors that might be able to modify the annotations but not likely to be easy ?
